### DISCUSSION TITLE
Advanced Foundry | Section-2 | part-11 | Pinata.cloud

### DISCUSSION THREAD

**Discussion by vigneshbalu96** [Cyfrin/foundry-full-course-cu#3070] [Q&A]
@2:54 mins - How do I actually ensure that when uploading my data to pinata.cloud would assure that other nodes actually pin my data?

**Comment by EngrPips** [comment]
Hello @vigneshbalu96, I can't remember correctly but I think Patrick touched on this.

**Comment by KMean** [comment]
@vigneshbalu96 I believe pinata automatically pins your data on their IPFS nodes when you upload it. To ensure widespread availability, you can also use additional pinning services like, [Infura IPFS](https://www.infura.io/product/ipfs), Web3.Storage, Filecoin etc. You can also run your own IPFS node and pin the data there. By doing this, your node becomes part of the IPFS network, improving redundancy.

**Reply by EngrPips** [reply]
Interesting heads-up.
